src/client.ts

```diff
@@ -235,31 +231,18 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     this.isClientReady = !!configWithClientInfo.datafile;
     this.isUsingSdkKey = !!configWithClientInfo.sdkKey;
 
-    this.userPromise = new Promise(resolve => {
```

Comment:
I cannot follow all the dependences, but it looks like you simplified a lot. Looks great! 
Just want to make sure this does not break the case for async user setting.

Reply:
You and me both Sir.  It took too long to understand the Promise web. I'd love to unravel this more but I would rather not scope creep past the GA release.

Yes, the goal code allows for latent setting of the user in the user context leaning on the VUID in the absence of a real userId. 

I have a devcontainer ready and happy to demo the effect.